crates/http/src/routes.rs

```diff
@@ -314,7 +332,10 @@ mod route_tests {
 
         let r = Router { routes };
 
-        assert_eq!(r.route("/one/two/three/four")?, "one_wildcard".to_string());
+        assert_eq!(
```

Comment:
This covers the "longer beats shorter" case but we also need one for "exact beats wildcard."

Reply:
Ah yeah, good catch!

Reply:
@itowlson updated with new test!